Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -3898,6 +3899,44 @@ Optional<String> Element::lang() const
     return maybe_lang.release_value();
 }
 
+// https://drafts.csswg.org/css-images-4/#element-not-rendered
+bool Element::not_rendered() const
+{
+    // An element is not rendered if it does not have an associated box.
+    if (!layout_node() || !paintable_box())
+        return true;
+
+    return false;
+}
+
+// https://drafts.csswg.org/css-view-transitions-1/#document-scoped-view-transition-name
+Optional<FlyString> Element::document_scoped_view_transition_name()
+{
+    // To get the document-scoped view transition name for an Element element:
+
+    // 1. Let scopedViewTransitionName be the computed value of view-transition-name for element.
+    auto scoped_view_transition_name = computed_properties()->view_transition_name();
+
+    // 2. If scopedViewTransitionName is associated with element’s node document, then return
+    //    scopedViewTransitionName.
+    // FIXME: Properly handle tree-scoping of the name here.
+    //        (see https://drafts.csswg.org/css-view-transitions-1/#propdef-view-transition-name , "Each view transition name is a tree-scoped name.")
+    if (true) {
+        return scoped_view_transition_name;
+    }
+
+    // 3. Otherwise, return none.
+    return {};
+}
+
+// https://drafts.csswg.org/css-view-transitions-1/#capture-the-image
+// To capture the image given an element element, perform the following steps. They return an image.
+RefPtr<Gfx::Bitmap> Element::capture_the_image()
+{
+    // FIXME: Actually implement this.
+    return MUST(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, Gfx::AlphaType::Premultiplied, Gfx::IntSize(1, 1)));
```

Comment:
not important at this point, but `capture_the_image()` will likely have to return `ImmutableBitmap` or `PaintingSurface` so we don't have to read back data from gpu memory.